contracts/OgvStaking.sol

```diff
@@ -151,6 +156,12 @@ contract OgvStaking is ERC20Votes {
         );
         _mint(to, points);
         ogv.transferFrom(msg.sender, address(this), amount); // Important that it's sender
+
+        if (!hasManualDelegation[msg.sender] && delegates(to) == address(0)) {
```

Comment:
🛑 Using `message.sender` in this line is a bug?

The scenario we are trying to avoid is:

- An address renounces delegation.
- A third party gives that address a tiny stake, and reenables delegation. 

If we do the `hasManualDelegation`check on the sender of the stake, not the receiver, even if the receiver has revoked voting, as long as the attacker has not delegated, the attacker can still complete the attack.

This test fails in forge.

```javascript
    function testRenounceAttack() external {
        // Alice can vote, because she is staked
        assertEq(staking.getVotes(alice), 1 * POINTS, "can vote after staking");

        // Alice renounces voting.
        vm.prank(alice);
        staking.delegate(address(0));

        // Attacker attacks
        vm.startPrank(attacker);
        ogv.mint(attacker, 1 ether);
        ogv.approve(address(staking), 1 ether);
        staking.stake(1 ether, 100 days, alice);
        vm.stopPrank();

        vm.roll(2);

        // Alice should still have renounced voting
        assertEq(staking.getVotes(alice), 0, "can't vot after renouncing");
    }
 ```

Reply:
I have overlooked this 😞 I think you are right @DanielVF it should be `!hasManualDelegation[to]` in there.

Reply:
Thanks for catching this, will fix it now

Reply:
PR #377 should fix this